### PR TITLE
Port infrastructure improvements from master to dev/release/2.0.0

### PR DESCRIPTION
**Pull request by nguerrera**
Porting the following to the new 2.0.x infrastructure branch from master:

#7352 Update solution to reflect current state
#7530 Eliminate warnings (just that subset, not the part of the PR that updated core-setup)
#7460 Use REST API to upload to feed
#7571 Logging for feed upload
#7522 Rearrange output folders to fix incremental build
#7594 Fix signing after output folders moved

### CODE REVIEW

**Comment by nguerrera** [comment]
@mmitche is there a delay before newly added branches trigger ci?

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
@nguerrera Good question...checking.

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
@nguerrera Looks like just a configuration error post-upgrade.  Should trigger in a couple

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by mmitche** [comment]
@dotnet-bot test this please

**Comment by livarcocc** [comment]
@dotnet-bot Test Windows_NT x86 Debug Build

File: build_projects/dotnet-cli-build/dotnet-cli-build.csproj [hunk 1]
```diff
@@ -24,10 +23,12 @@
     <PackageReference Include="WindowsAzure.Storage" Version="7.2.1" />
     <PackageReference Include="NuGet.CommandLine.XPlat" Version="$(CLI_NuGet_Version)" />
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(CLI_MSBuild_Version)" />
+    <PackageReference Include="Microsoft.Build.Runtime" Version="$(CLI_MSBuild_Version)" />
+
     <!-- This dependency was added due to an issue in restore where a lower version of this package coming from nuget.commandline.xplat
     led to an error. This is tracked as NuGet issue : https://github.com/NuGet/Home/issues/4213 -->
     <PackageReference Include="Microsoft.Build.Framework" Version="$(CLI_MSBuild_Version)" />
-    <PackageReference Include="Microsoft.DotNet.PlatformAbstractions" Version="$(PlatformAbstractionsVersion)" />
+    <PackageReference Include="Microsoft.DotNet.PlatformAbstractions" Version="2.0.0" />
```
Comment (livarcocc) [inline build_projects/dotnet-cli-build/dotnet-cli-build.csproj]: Do we know why we are pinning this?

Comment (nguerrera) [reply build_projects/dotnet-cli-build/dotnet-cli-build.csproj]: It blocked the ingestion of core-setup 2.1. Basically, build projectscan't run on stage0 2.0 if they ref 2.1. There is no need for the build projects to reference the version the product will build against. I think we should pin all of the versions here to something stable unless we actually require a new feature in a build project.

Comment (nguerrera) [reply build_projects/dotnet-cli-build/dotnet-cli-build.csproj]: We could put it back now but then we'll hit the same bug on the next core-setup version bump.

**Comment by livarcocc** [comment]
@mmitche 

```
Time Elapsed 00:21:47.54
11:47:19 [xUnit] [INFO] - Starting to record.
11:47:19 [xUnit] [INFO] - Processing MSTest-Version N/A (default)
11:47:24 [xUnit] [INFO] - [MSTest-Version N/A (default)] - 32 test report file(s) were found with the pattern '**/*.trx' relative to 'D:\j\workspace\debug_windows---bb0a5091' for the testing framework 'MSTest-Version N/A (default)'.
11:47:50 [xUnit] [INFO] - Check 'Failed Tests' threshold.
11:47:50 [xUnit] [INFO] - Check 'Skipped Tests' threshold.
11:47:50 [xUnit] [INFO] - Setting the build status to SUCCESS
11:47:50 [xUnit] [INFO] - Stopping recording.
11:47:50 Notifying endpoint with url 'helix-int-notification-url'
11:47:50 ERROR: Failed to notify endpoint with url 'helix-int-notification-url'
11:47:50 java.net.MalformedURLException: no protocol: helix-int-notification-url
11:47:50 	at java.net.URL.<init>(URL.java:593)
11:47:50 	at java.net.URL.<init>(URL.java:490)
11:47:50 	at java.net.URL.<init>(URL.java:439)
11:47:50 	at com.tikal.hudson.plugins.notification.Protocol$3.send(Protocol.java:53)
11:47:50 	at com.tikal.hudson.plugins.notification.Phase.handle(Phase.java:66)
11:47:50 	at com.tikal.hudson.plugins.notification.JobListener.onCompleted(JobListener.java:39)
11:47:50 	at hudson.model.listeners.RunListener.fireCompleted(RunListener.java:211)
11:47:50 	at hudson.model.Run.execute(Run.java:1782)
11:47:50 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
11:47:50 	at hudson.model.ResourceController.execute(ResourceController.java:97)
11:47:50 	at hudson.model.Executor.run(Executor.java:419)
11:47:50 Failed to notify endpoint with url 'helix-int-notification-url' - java.net.MalformedURLException: no protocol: helix-int-notification-url
11:47:50 Notifying endpoint with url 'helix-prod-notification-url'
11:47:50 ERROR: Failed to notify endpoint with url 'helix-prod-notification-url'
11:47:50 java.net.MalformedURLException: no protocol: helix-prod-notification-url
11:47:50 	at java.net.URL.<init>(URL.java:593)
11:47:50 	at java.net.URL.<init>(URL.java:490)
11:47:50 	at java.net.URL.<init>(URL.java:439)
11:47:50 	at com.tikal.hudson.plugins.notification.Protocol$3.send(Protocol.java:53)
11:47:50 	at com.tikal.hudson.plugins.notification.Phase.handle(Phase.java:66)
11:47:50 	at com.tikal.hudson.plugins.notification.JobListener.onCompleted(JobListener.java:39)
11:47:50 	at hudson.model.listeners.RunListener.fireCompleted(RunListener.java:211)
11:47:50 	at hudson.model.Run.execute(Run.java:1782)
11:47:50 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
11:47:50 	at hudson.model.ResourceController.execute(ResourceController.java:97)
11:47:50 	at hudson.model.Executor.run(Executor.java:419)
11:47:50 Failed to notify endpoint with url 'helix-prod-notification-url' - java.net.MalformedURLException: no protocol: helix-prod-notification-url
11:47:50 Notifying endpoint with url 'legacy-notification-url'
11:47:50 ERROR: Failed to notify endpoint with url 'legacy-notification-url'
11:47:50 java.net.MalformedURLException: no protocol: legacy-notification-url
11:47:50 	at java.net.URL.<init>(URL.java:593)
11:47:50 	at java.net.URL.<init>(URL.java:490)
11:47:50 	at java.net.URL.<init>(URL.java:439)
11:47:50 	at com.tikal.hudson.plugins.notification.Protocol$3.send(Protocol.java:53)
11:47:50 	at com.tikal.hudson.plugins.notification.Phase.handle(Phase.java:66)
11:47:50 	at com.tikal.hudson.plugins.notification.JobListener.onCompleted(JobListener.java:39)
11:47:50 	at hudson.model.listeners.RunListener.fireCompleted(RunListener.java:211)
11:47:50 	at hudson.model.Run.execute(Run.java:1782)
11:47:50 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
11:47:50 	at hudson.model.ResourceController.execute(ResourceController.java:97)
11:47:50 	at hudson.model.Executor.run(Executor.java:419)
11:47:50 Failed to notify endpoint with url 'legacy-notification-url' - java.net.MalformedURLException: no protocol: legacy-notification-url
11:47:50 Setting status of a652e6a4fd6733a69b0bd8c99e55372ff1be0974 to SUCCESS with url 
```

**Comment by mmitche** [comment]
@livarcocc Please ignore those errors for now.

**Pull request merged**